hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpCommand.java

```diff
@@ -56,14 +55,25 @@
     protected final String uri;
     protected ByteBuffer response;
     protected boolean nextLine;
+    protected final RestCallExecutionListener listener;
 
-
-    public HttpCommand(TextCommandConstants.TextCommandType type, String uri) {
+    public HttpCommand(TextCommandConstants.TextCommandType type, String uri, ServerConnection connection) {
         super(type);
         this.uri = uri;
         // the command line was parsed already, let's start with clear next line
         this.nextLine = true;
+        this.listener = connection.getConnectionManager().getServer().getContext()
```

Comment:
This line tells me that there is something wrong with the design. 
Just to access `createRestCallExecutionListener` method we are calling 5 different levels. 

Ideally `RestCallExecutionListener` should be created outside of the HttpCommand constructor and passed here. 
Alternatively, (sounds more correct to me) `RestCallExecutionListener` should not be a field of HttpCommand. Logically, only state that HttpCommand should carry looks like `RestCallExecution executionDetails` . Others does not have state. We can make `RestCallExecutionListener` stateless and pass it to related methods instead. When we design it like this, it looks more suitable to call listener outside of the httpCommand.  

I started refactoring the code to understand what I will want is feasible and I have ended up with this pr. 
https://github.com/erosb/hazelcast/pull/24
Please take a look. The advantages is that 
1. there is no extra object(`RestCallExecutionListener`) per command anymore. 
2. We don't have a constructor trying to access something central. 
3. I also removed HTTPMethod and using TextCommandType instead since it started to be more obvous that there is duplication while I am refactoring.